### ISSUE TITLE
[ENH] printpreview is not language agnostic

### ISSUE THREAD

**Issue by dubourg** [ipython-contrib/IPython-notebook-extensions#675]
```javascript
IPython.notebook.kernel.execute("call_to_nbconvert_depends_on_kernel_language")
```
Do you have any idea how we could make this nbextension compatible with non-python kernels?
I don't know much about javascript but wouldn't it be possible to call nbconvert in shell/bash/batch directly (bypassing the kernel) ?
cc @juhasch @fdeheeger

**Comment by janschulz** [comment]
IMO this highlights a need for a channel to the notebook server itself.

**Comment by juhasch** [comment]
Not sure what the (new) security implications of calling the notebook server itself are.

My preferred way would be to allow the frontend to specify a configuration (pre-/postprocessor, templates, exporter) when it calls nbconvert.

**Comment by dubourg** [comment]
@juhasch yes indeed! For my specific concern here, I'd like to be able to choose whether I want to display a cell input/output or not. This could even simply use cells metadata. Currently we (mis)use slideshow metadata and a custom css in order to do that.

**Comment by jcb91** [comment]
> My preferred way would be to allow the frontend to specify a configuration (pre-/postprocessor, templates, exporter) when it calls nbconvert.

yes, seems like the best option. This would require a server extension to provide an API accessible from javascript, similar to existing nbconvert filehandlers from `notebook.nbconvert.handlers`, but with a few more URL parameters to account for nbconvert options, I guess...

**Comment by juhasch** [comment]
Yes, and I guess that would not be too difficult to do.

**Comment by juhasch** [comment]
See https://github.com/jupyter/notebook/issues/1629

**Comment by jcb91** [comment]
alternatively, assuming the kernel supports the generic magic syntax of `!arbitrary shell command`, that could be used?
